### ISSUE TITLE
NDCG calculation that produces score identical to the sklearn yet faster

### ISSUE THREAD

**Issue by AnastasiiaShukhova** [abstractqqq/polars_ds_extension#346]
This library and polars framework in general misses some ranking metrics like NDCG. See the implementation in the sklearn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ndcg_score.html

I have implemented NDCG metric calculation in polars that matches sklearn evaluation results and opened a PR to add it. @abstractqqq would you please have a look? I will post the evaluation results and speed increase later today.

**Comment by abstractqqq** [comment]
Yes of course! 

Fork the library first. Then clone the fork to local.

Add your code. For metrics, there should be a metrics.py somewhere in exprs. Commit the code to the fork first.

Then in github you should be able to see a merge with upstream options.

If there is anything you are not sure about, feel free to let me know.
